src/sfdp.c

```diff
@@ -143,6 +144,57 @@ static int read_jedec_flash_parameter(FLASHER_SPI_FLASH_T *ptFlash, unsigned lon
 	{
 		hexdump(uSfdpData.auc, sizeof(uSfdpData.auc));
 
+		/** Read erase instruction entries from JEDEC Basic Flash Parametes
+		 *  Read starting from DWORD 8, there are 4 erase entries each with:
+		 *  8-Bit for the size in 2^N format
+		 *  8-Bit for the OpCode
+		 */
+		const unsigned char DWORD_8_START = 7*4; /* Nr of DWords starts at 1*/
+		unsigned int nrEraseOps = 0; /* Required for sorting */
+
+		/* Get erase operations from SFDP data */
+		for(int opNr = 0; opNr < FLASHER_SPI_NR_ERASE_INSTRUCTIONS; opNr++){
```

Comment:
Do we have to take into account the case that there's a hole in the table, e.g. entries 0 and 2 are occupied?

Reply:
You are correct in that there is no requirement for the fields to be filled in order, as far as I can tell.
There are already mechanisms to account for this:
- We always read all 4 instructions, if we find an empty instruction save it as 0 size, 0 opcode
- Sort the entries in ptFlash->tSpiErase ordered by erase size, starting with the smallest non-0 sized operation at element 0 (line 170)

However I just noticed that the sorting loop in line 171 will not sort correctly if the 0-Entries are in the middle.
I'll address this and get back to you.

Reply:
First clear the whole list, then copy only those entries that are non-zero:
	if(shift != 0 && shift < sizeof(unsigned long long)*8){
		ptFlash->tSpiErase[nrEraseOps].Size = 1U << shift;
		ptFlash->tSpiErase[nrEraseOps].OpCode = opCode;
		nrEraseOps++;
	}